routers/api/packages/container/manifest.go

```diff
@@ -282,6 +289,17 @@ func processImageManifestIndex(mci *manifestCreationInfo, buf *packages_module.H
 	return manifestDigest, nil
 }
 
+func notifyPackageCreate(doer *user_model.User, pv *packages_model.PackageVersion) error {
+	pd, err := packages_model.GetPackageDescriptor(db.DefaultContext, pv)
+	if err != nil {
+		return err
+	}
+
+	notification.NotifyPackageCreate(db.DefaultContext, doer, pd)
```

Comment:
I don't quite like using `db.DefaultContext` in `routers`, but I'll allow it for now.

Reply:
How about just use context ?

Reply:
The notification should not use a transaction context. That's the reason why I use the `DefaultContext`.

Reply:
If the `pv` is created inside an uncommited transaction, then does `GetPackageDescriptor(db.DefaultContext, pv)` work? I haven't read the code, just a question.

Reply:
Yes, the only "problem" is if satelite data (the owner/creator, repository, ...) referenced by that package version is deleted between the commit and the `GetPackageDescriptor` call. But in that case `GetPackageDescriptor` just returns an error.

Reply:
Hmm, I am not sure about what the "yes" means ....

* If the "yes" means that: it works, because there is no uncommited transaction when calling notifyPackageCreate, then no problem.
* If the "yes" means that: `GetPackageDescriptor(db.DefaultContext, pv)` could read uncommited data, then I do not think so. Uncommited transaction data couldn't be read by other transactions.

Maybe the stable logic could be: `GetPackageDescriptor(dbCtx)` and `NotifyPackageCreate(DefaultContext)`, then there won't be problem.

Reply:
"Yes" as "there is no uncommitted transaction". A lot of the notify methods collect their data inside the method. In this case the data-collections happens before the notification.

Reply:
For sqlite3, if a transaction hasn't been committed or rollback, starting another transaction in the same go routine will result in a deadlock.

Reply:
I have read code   after the question. The  notifyPackageCreate is called after the transaction commit. So no problem